### PR TITLE
Add NePaginator component

### PR DESCRIPTION
**Pull request by kevincela**
- Add NePaginator component and relative stories

### CODE REVIEW

File: src/components/NePaginator.vue [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+import { range } from 'lodash-es'
+
+export type NePaginatorProps = {
+  currentPage: number
+  totalPages: number
+  previousLabel: string
+  nextLabel: string
+}
+
+const props = defineProps<NePaginatorProps>()
```
Comment (andre8244) [inline src/components/NePaginator.vue:17]: When using Typescript to declare props, the linter sometimes find it hard to do type checking, e.g. detecting required props not passed to the component. I suggest to use the  "old standard" way to define props, specifying `type` and `required` or `default` for every prop

File: src/components/NePaginator.vue [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  <nav>
+    <ul class="flex h-10 items-center -space-x-px text-base">
+      <li>
+        <button
+          :disabled="currentPage === 1"
+          class="ms-0 flex h-10 items-center justify-center rounded-s-lg border border-e-0 border-gray-300 bg-white px-4 leading-tight text-gray-500 hover:bg-gray-100 hover:text-gray-700 dark:border-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
+          @click="navigateToPage(currentPage - 1)"
+        >
+          <span class="sr-only">{{ previousLabel }}</span>
+          <svg
```
Comment (andre8244) [inline src/components/NePaginator.vue:58]: Let's use Fontawesome `chevron-left` icon, for the sake of UI consistency

File: src/components/NePaginator.vue [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        </li>
+      </template>
+      <li>
+        <button
+          :disabled="currentPage === totalPages"
+          class="flex h-10 items-center justify-center rounded-e-lg border border-gray-300 bg-white px-4 leading-tight text-gray-500 hover:bg-gray-100 hover:text-gray-700 dark:border-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
+          @click="navigateToPage(currentPage + 1)"
+        >
+          <span class="sr-only">{{ nextLabel }}</span>
+          <svg
```
Comment (andre8244) [inline src/components/NePaginator.vue:148]: Let's use Fontawesome `chevron-right` icon, for the sake of UI consistency

File: src/components/NePaginator.vue (hunk 1 above)
Comment (kevincela) [reply src/components/NePaginator.vue]: Sure, though I've still left the `NePaginatorProps` type and export for all outer components which want to use the type of the paginator's props!

**Review by andre8244 — APPROVED** [review]

**Pull request merged**